underscore.js

```diff
@@ -1223,7 +1223,7 @@
         // Object(NaN) is equivalent to NaN.
         if (+a !== +a) return +b !== +b;
         // An `egal` comparison is performed for other numeric values.
-        return +a === 0 ? 1 / +a === 1 / b : +a === +b;
+        return +a === 0 && +b === 0 ? 1 / +a === 1 / b : +a === +b;
```

Comment:
This just collapses the entire expression to `+a === +b`, since to enter the first branch, both `+a` and `+b` must already have been found to exactly equal `0`. You can preserve the original fuzziness and still make the comparison symmetric by using a disjunction instead of a conjunction:

```suggestion
        return +a === 0 || +b === 0 ? 1 / +a === 1 / +b : +a === +b;
```

It may actually be worth considering to also allow fuzziness when both values are close to zero but neither is exactly zero. This seems more consistent as it doesn't treat zero as a special case:

```suggestion
        return +a === +b || 1 / +a === 1 / +b;
```

However, maybe there is actually something to say for making the comparison symmetrically exact instead of symmetrically fuzzy:

```suggestion
        return +a === +b;
```

Reply:
Actually this is not really true. The initial equality check distinguish `0` from `-0`. It is not the case of the suggested change:
```diff
---        return +a === 0 && +b === 0 ? 1 / +a === 1 / b : +a === +b;
+++        return +a === +b;
```

As in JavaScript `-0 === 0`.

If I'm not wrong the final fix done in jest was to switch to `Object.is(+a, +b)` but I'll need to double-check to confirm that.

Reply:
Good point. So actually we have *four* symmetric options, from most to least fuzzy:

(1) `+0 === -0` and values that approach zero from the same side are equivalent.

```js
+a === +b || 1 / +a === 1 / +b
```

(2) `+0 !== -0` and values that approach zero from the same side are equivalent.

```js
+a === 0 || +b === 0 ? 1 / +a === 1 / +b : +a === +b
```

(3) `+0 === -0`, otherwise exact comparison.

```js
+a === +b
```

(4) `+0 !== -0`, entirely like `Object.is` but compatible with IE.

```js
+a === 0 && +b === 0 ? 1 / +a === 1 / +b : +a === +b
```